wordpress-mobile/WordPress-Android
# [Reader] Implement announcement card

Fixes #20621 

<p float="left">
<img width="350" alt="Screenshot 2024-05-17 at 10 25 32 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/b8e5bba6-4026-4a6b-a3ba-50d1a47d44b3">

<img width="350" alt="Screenshot 2024-05-17 at 10 25 25 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/59647aa6-177a-4b6e-b846-66d95a520990">

https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/fca2282a-ad09-47a1-b6c2-f5b2de2b127e

</p>

-----

## To Test:

- Install JP and log in.
- Make sure the `reader_announcement_card` feature flag is *disabled*.
- Open Reader.
- 🔍 Verify the announcement card is not shown in any feed.
- Enable `reader_announcement_card` feature flag.
- Open Reader.
- 🔍 Verify the announcement card appears as expected in all the feeds.
- 🔍 Tap "Done" button: make sure the announcement card disappears and is never shown again (unless the app data is cleared).

-----

## Regression Notes

1. Potential unintended areas of impact

    - Reader feeds layouts since I had to change a bit the `reader_fragment_layout` file

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual and unit testing.

3. What automated tests I added (or what prevented me from doing so)

    - Updated `ReaderViewModelTest.kt`.

-----

## PR Submission Checklist:


-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):
